discovery-provider/src/tasks/index.py

```diff
@@ -354,6 +373,121 @@ def save_and_get_skip_tx_hash(session, redis):
     return None
 
 
+def group_transaction_by_type(tx_type_to_grouped_lists_map, tx, tx_receipt):
+    tx_target_contract_address = tx["to"]
+    for tx_type, tx_list in tx_type_to_grouped_lists_map.items():
+        tx_is_type = tx_target_contract_address == get_contract_addresses()[tx_type]
+        if tx_is_type:
+            contract_name = helpers.snake_to_pascal(tx_type)
+            logger.info(
+                f"index.py | {contract_name} contract addr: {tx_target_contract_address}"
+                f" tx from block - {tx}, receipt - {tx_receipt}"
+            )
+            tx_list.append(tx_receipt)
+            break
+
+
+def add_indexed_block_to_db(db_session, block):
+    web3 = update_task.web3
+    current_block_query = db_session.query(Block).filter_by(is_current=True)
+
+    # Without this check we may end up duplicating an insert operation
+    block_model = Block(
+        blockhash=web3.toHex(block.hash),
+        parenthash=web3.toHex(block.parentHash),
+        number=block.number,
+        is_current=True,
+    )
+
+    # Update blocks table after
+    assert current_block_query.count() == 1, "Expected single row marked as current"
+
+    previous_block = current_block_query.first()
+    previous_block.is_current = False
+    db_session.add(block_model)
+
+
+def add_indexed_block_to_redis(block, redis):
+    redis.set(most_recent_indexed_block_redis_key, block.number)
+    redis.set(most_recent_indexed_block_hash_redis_key, block.hash.hex())
+
+
+def bulk_process_state_changes(
+    main_indexing_task,
+    session,
+    ipfs_metadata,
+    blacklisted_cids,
+    tx_type_to_grouped_lists_map,
+    block,
+    redis,
+):
+    block_number, block_hash, block_timestamp = itemgetter(
+        "number", "hash", "timestamp"
+    )(block)
+
+    TX_TYPE_TO_HANDLER_MAP = {
```

Comment:
feels to me that this should be a const outside of this fn

Reply:
Sounds good! Would it be too obfuscated living in `constants.py`? Do you think we should just keep it at the top of this file?

Reply:
top o the file seems good :)